### DISCUSSION TITLE
How to log in to SharePoint online and test the Webparts / Command sets

### DISCUSSION THREAD

**Discussion by necik11** [cypress-io/cypress#27615] [Questions and Help]
I tried looking for some tutorials, but either they are old, still with cypress 9 or a lot of other libraries are used which are not suits and making just hardcore solution...
Isn't there a simple way to log in?

This seems its will be right one: [Tutorial on Azure Active Directory Authentication
](https://docs.cypress.io/guides/end-to-end-testing/azure-active-directory-authentication) -
login works for me, but still when` cy.visit("mytenant.sharepoint.com")`
i have got login.microsoft.com ... i try even origin... some different attributes cypres.config.json and still some problems... for instance deadlock with iframe call or some other problem....
Give me tip i will try whatever but after what i have seen there is no approach how to achieve - login to SharePoint online site and for sample `  cy.title().should('eq', "whatever");`

**Comment by portyu9** [comment]
Hi @necik11 

For current Cypress, the supported approach is the same general pattern as Cypress's Azure AD example:

1. let SharePoint redirect the top-level page to Microsoft login;
2. perform the Microsoft-origin commands inside `cy.origin('login.microsoftonline.com', ...)`;
3. cache the authenticated state with `cy.session()`.

The important distinction is **top-level redirect vs iframe**.

`cy.origin()` works when the browser navigates to `login.microsoftonline.com`. It does **not** let Cypress automate a Microsoft login form embedded in a cross-origin iframe.

Also, Cypress cannot bypass Microsoft/tenant authentication policies. Conditional Access, MFA, federated identity, bot protection, etc. can make fully automated interactive login unsuitable. For automation, a dedicated test tenant/account with an authentication policy explicitly intended for testing is much more reliable.

I would follow the current Cypress Azure AD authentication example rather than older Cypress 9 tutorials. Cache the successful login with `cy.session()` so you don't run through Microsoft authentication for every test.

If `cy.visit('https://<tenant>.sharepoint.com')` redirects at the top level, `cy.origin()` is the right direction. If what you are blocked on is specifically a login UI inside a cross-origin iframe, that is a different limitation.

Please mark this answer as accepted if it helped, thank you.
